Tests/ClientRuntimeTests/NetworkingTests/AsyncThrowingStreamReaderTests.swift

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+import XCTest
+@testable import ClientRuntime
+
+final class AsyncThrowingStreamReaderTests: XCTestCase {
+    func testRead() {
```

Comment:
Can we add another test that verifies the locking behavior by spinning up multiple concurrent calls to `read` and verifying the output is as expected?

Reply:
now we use actor, we don't want to test actor.